lir/aggregation.py

```diff
@@ -48,36 +52,63 @@ def close(self) -> None:  # noqa: B027
 class AggregatePlot(Aggregation):
     """Aggregation that generates plots by repeatedly calling a plotting function."""
 
-    def __init__(self, plot_function: Callable, output_dir: str) -> None:
-        super().__init__()
-
-        self.f = plot_function
-        self.dir = output_dir
-        self.plot_type = plot_function.__name__
-        self._fig, self._ax = plt.subplots(figsize=(10, 8))
-        self._canvas = Canvas(self._ax)
+    def __init__(self, plot_fn: Callable, plot_name: str, output_dir: Path | None = None) -> None:
+        self.output_path = output_dir
+        self.plot_fn = plot_fn
+        self.plot_name = plot_name
 
     def report(self, data: AggregationData) -> None:
-        self._canvas.plot(
-            [],
-            [],
-            marker='None',
-            linestyle='None',
-            color='white',  # This is necessary to avoid matplotlib from cycling through colours
-            label=', '.join(f'{k}={v}' for k, v in data.parameters.items()),
-        )  # Dummy plot to add legend entry
+        """Plot the data when new results are available."""
+        fig, ax = plt.subplots()
 
-        self.f(None, data.llrdata, self._canvas)
+        llrdata = data.llrdata
+        parameters = data.parameters
 
-    def close(self) -> None:
-        """Generate and save each plot after all results have been reported."""
-        self._ax.set_title(f'Aggregated {self.plot_type}')
-        self._fig.savefig(f'{self.dir}/aggregated_{self.plot_type}.png')
+        self.plot_fn(llrdata=llrdata, ax=ax)
+
+        # Only save the figure when an output path is provided.
+        if self.output_path is not None:
+            dir_name = self.output_path
+            param_string = '__'.join(f'{k}={v}' for k, v in parameters.items())
+            file_name = dir_name / param_string / f'{self.plot_name}.png'
+            dir_name.mkdir(exist_ok=True, parents=True)
```

Comment:
Ik vind het niet zo mooi om hier te vertrouwen op een eerder gemaakte directory voor param_string. Eigenlijk sowieso niet om op twee plaatsen dezelfde directorynaam samen te stellen. Ik stel voor nu de slash vervangen door underscore en dan gauw mergen :-)

```suggestion
            file_name = dir_name / f'{param_string}_{self.plot_name}.png'
            dir_name.mkdir(exist_ok=True, parents=True)
```